### PR TITLE
Userparamlimits

### PR DESCRIPTION
**Pull request by alihadi-mazeh**
Added character limits and validations for user params

## Description

Added character limits to user name, password, and email and a validation for language with their corresponding validation.
Added validation for character limits on the front end for the same params

### CODE REVIEW

File: app/javascript/helpers/forms/SignupFormHelpers.jsx [hunk 1]
```diff
@@ -3,9 +3,14 @@ import { yupResolver } from '@hookform/resolvers/yup';
...
   // TODO: amir - Revisit validations.
-  name: yup.string().required('Please enter a full name.'),
-  email: yup.string().required('Please enter an email.').email('Entered value does not match email format.'),
-  password: yup.string()
+  name: yup.string().required('Please enter a full name.').min(2, 'Name must be at least 2 characters long')
+    .max(255, 'Name must be at most 255 characters long'),
+
+  email: yup.string().required('Please enter an email.').email('Entered value does not match email format.')
+    .min(6, 'Email must be at least 6 characters long')
+    .max(255, 'Email must be at most 5 characters long'),
```
Comment (scouillard) [inline app/javascript/helpers/forms/SignupFormHelpers.jsx:11]: Typo here, should be 255 not 5

File: app/javascript/helpers/forms/SignupFormHelpers.jsx [hunk 2]
```diff
@@ -3,9 +3,14 @@ import { yupResolver } from '@hookform/resolvers/yup';
 
 const validationSchema = yup.object({
   // TODO: amir - Revisit validations.
-  name: yup.string().required('Please enter a full name.'),
-  email: yup.string().required('Please enter an email.').email('Entered value does not match email format.'),
-  password: yup.string()
+  name: yup.string().required('Please enter a full name.').min(2, 'Name must be at least 2 characters long')
```
Comment (farhatahmad) [inline app/javascript/helpers/forms/SignupFormHelpers.jsx:6]: To make it a bit easier to read, let's put each validation on a separate line
```
name: yup.string().required('Please enter a full name.')
     .min(2, 'Name must be at least 2 characters long')
     .max(255, 'Name must be at most 255 characters long'),
```

File: app/models/user.rb [hunk 3]
```diff
@@ -47,6 +47,11 @@ class User < ApplicationRecord
   validates :verification_digest, uniqueness: true, if: :verification_digest?
   validates :session_token, presence: true, uniqueness: true
   validates :session_expiry, presence: true
+  validates :language, presence: true
+
+  validates :name, length: { minimum: 2, maximum: 255 }
+  validates :email, length: { minimum: 5, maximum: 255 }
+  validates :password, length: { maximum: 255 }
```
Comment (farhatahmad) [inline app/models/user.rb:54]: We already have `validates` statements for these fields (above) - can we just merge them in?

File: spec/models/room_spec.rb [hunk 4]
```diff
@@ -11,6 +11,8 @@
     it { is_expected.to belong_to(:user) }
     it { is_expected.to have_many(:recordings).dependent(:destroy) }
     it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_length_of(:name).is_at_least(2) }
```
Comment (farhatahmad) [inline spec/models/room_spec.rb:14]: Did you try doing it in one line? `it { is_expected.to validate_length_of(:name).is_at_least(2).is_at_most(255) }

File: app/javascript/helpers/forms/SignupFormHelpers.jsx (hunk 1 above)
Comment (alihadi-mazeh) [reply app/javascript/helpers/forms/SignupFormHelpers.jsx:11]: Good catch

File: spec/models/room_spec.rb (hunk 4 above)
Comment (alihadi-mazeh) [reply spec/models/room_spec.rb]: Changes applied :heavy_check_mark:

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -22,6 +22,15 @@
 
     it { is_expected.to validate_presence_of(:session_token) }
     it { is_expected.to validate_presence_of(:session_expiry) }
+    it { is_expected.to validate_presence_of(:language) }
+
+    it { is_expected.to validate_length_of(:name).is_at_least(2) }
+    it { is_expected.to validate_length_of(:name).is_at_most(255) }
+
+    it { is_expected.to validate_length_of(:email).is_at_least(5) }
+    it { is_expected.to validate_length_of(:email).is_at_most(255) }
```
Comment (farhatahmad) [inline spec/models/user_spec.rb:31]: Don't forget to make the changes here as well. you can do it in one line

Comment (alihadi-mazeh) [reply spec/models/user_spec.rb:31]: Okay on it

File: app/models/user.rb [hunk 6]
```diff
@@ -22,17 +22,21 @@ class User < ApplicationRecord
...
   validates :provider, presence: true
   validates :status, presence: true
   validates :password,
             presence: true,
-            on: :create, unless: :external_id?
+            on: :create, unless: :external_id?,
+            length: { maximum: 255 }
 
   validates :password,
             format: %r{\A(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[`@%~!#£$\\^&*()\]\[+={}/|:;"'<>\-,.?_ ]).{8,}\z},
```
Comment (scouillard) [inline app/models/user.rb:42]: Could those be bundled as well?

**Review by farhatahmad — APPROVED** [review]

File: app/models/user.rb (hunk 6 above)
Comment (alihadi-mazeh) [reply app/models/user.rb:42]: bundle the two validates statements for the password?

**Review by farhatahmad — DISMISSED** [review]

File: app/models/user.rb (hunk 6 above)
Comment (alihadi-mazeh) [reply app/models/user.rb:42]: i think it's best to leave it for another PR

**Pull request merged**